src/distributed_ls/Euclid/Hash_i_dh.c

```diff
@@ -135,7 +135,8 @@ HYPRE_Int Hash_i_dhLookup(Hash_i_dh h, HYPRE_Int key)
 */
 
   for (i=0; i<size; ++i) {
-    idx = (start + i*inc) % size;
+    /* idx = (start + i*inc) % size; */
+    idx = (start + hypre_multmod(i, inc, size)) % size;
```

Comment:
@liruipeng is hypre_multmod replacing (i*inc) % size? If so should we be taking % size outside the parenthesis? Should it is idx = start % size + hypre_multmod(i, inc, size) ?

Reply:
@oseikuffuor1 
```
idx = (start + i*inc) % size
      = (start + (i*inc) % size) % size
      = (start + hypre_multmod(i, inc, size)) % size
```

Reply:
Got it. Thanks for the earlier reference and explanation.